pytorch_lattice/layers/rtl.py

```diff
@@ -0,0 +1,274 @@
+                    reverse=False,
+                )
+                groups[i] = sorted_indices
+
+        self._lattice_layers = {}
+        for monotonic_count, groups in monotonicity_groupings.items():
+            self._lattice_layers[monotonic_count] = (
+                Lattice(
+                    lattice_sizes=[self.lattice_size] * self.lattice_rank,
+                    output_min=self.output_min,
+                    output_max=self.output_max,
+                    kernel_init=self.kernel_init,
+                    monotonicities=[Monotonicity.INCREASING] * monotonic_count
+                    + [None] * (lattice_rank - monotonic_count),
+                    clip_inputs=self.clip_inputs,
+                    interpolation=self.interpolation,
+                    units=len(groups),
+                ),
+                groups,
+            )
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """Forward method computed by using forward methods of each lattice in RTL.
+
+        Args:
+            x: input tensor of feature values with shape `(batch_size, num_features)`.
+
+        Returns:
+            torch.Tensor containing the outputs of each lattice within RTL structure. If
+            `average_outputs == True`, then all outputs are averaged into a tensor of
+            shape `(batch_size, 1)`. If `average_outputs == False`, shape of tensor is
+            `(batch_size, num_lattices)`.
+        """
+        forward_results = []
+        for _, (lattice, group) in sorted(self._lattice_layers.items()):
+            if len(group) > 1:
+                lattice_input = torch.stack([x[:, idx] for idx in group], dim=-2)
+            else:
+                lattice_input = x[:, group[0]]
+            forward_results.append(lattice.forward(lattice_input))
+        result = torch.cat(forward_results, dim=-1)
+        if not self.average_outputs:
+            return result
+        result = torch.mean(result, dim=-1, keepdim=True)
+
+        return result
+
+    @torch.no_grad()
+    def output_monotonicities(self) -> List[Monotonicity]:
+        """Gives the monotonicities of the outputs of RTL.
+
+        Returns:
+            List of `Monotonicity` corresponding to each output of the RTL layer, in the
+            same order as outputs.
+        """
+        monotonicities = []
+        for monotonic_count, (lattice, _) in sorted(self._lattice_layers.items()):
+            if monotonic_count:
+                monotonicity = Monotonicity.INCREASING
+            else:
+                monotonicity = None
+            for _ in range(lattice.units):
+                monotonicities.append(monotonicity)
+
+        return monotonicities
+
+    @torch.no_grad()
+    def constrain(self) -> None:
+        """Enforces constraints for each lattice in RTL."""
+        for lattice, _ in self._lattice_layers.values():
+            lattice.constrain()
+
+    @torch.no_grad()
+    def assert_constraints(self, eps=1e-6) -> List[List[str]]:
```

Comment:
`eps` is missing type definition. Should read `eps: float = 1e-6`

Also, please update `List` to `list` here and everywhere else.